From ea2b20ed7ab3e83b8585e20d2febdbb5f2d356c1 Mon Sep 17 00:00:00 2001
From: Michele Calgaro <michele.calgaro@yahoo.it>
Date: Mon, 2 Aug 2021 11:24:50 +0900
Subject: [PATCH 1/4] Removed unused hardwareEvent code.

Initialization of TDEHardwareDevices() always happens before the signal
was connected. The constructor of TDEHardwareDevices() is the only place
that was triggering the signal emission, so the code was non functional.

Signed-off-by: Michele Calgaro <michele.calgaro@yahoo.it>
---
 tdenetworkmanager/src/devicetraycomponent.cpp            | 2 --
 tdenetworkmanager/src/tdenetman-tray.cpp                 | 9 ---------
 tdenetworkmanager/src/tdenetman-tray.h                   | 1 -
 tdenetworkmanager/src/tdenetman-wireless_device_tray.cpp | 3 ---
 4 files changed, 15 deletions(-)

diff --git a/tdenetworkmanager/src/devicetraycomponent.cpp b/tdenetworkmanager/src/devicetraycomponent.cpp
index 5f1f26e..085435c 100644
--- a/tdenetworkmanager/src/devicetraycomponent.cpp
+++ b/tdenetworkmanager/src/devicetraycomponent.cpp
@@ -74,8 +74,6 @@ DeviceTrayComponent::DeviceTrayComponent(TQString device, KSystemTray * parent,
 
 	TDEHardwareDevices* hwdevices = TDEGlobal::hardwareDevices();
 
-	// Listen for hardware change events
-	connect(hwdevices, TQT_SIGNAL(hardwareEvent(TDEHardwareEvent::TDEHardwareEvent, TQString)), this, TQT_SLOT(tdeHardwareEventHandler(TDEHardwareEvent::TDEHardwareEvent, TQString)));
 	// Trigger an initial hardware information update
 	tdeHardwareEventHandler(TDEHardwareEvent::HardwareListModified, TQString::null);
 
diff --git a/tdenetworkmanager/src/tdenetman-tray.cpp b/tdenetworkmanager/src/tdenetman-tray.cpp
index 9427679..173464f 100644
--- a/tdenetworkmanager/src/tdenetman-tray.cpp
+++ b/tdenetworkmanager/src/tdenetman-tray.cpp
@@ -963,14 +963,6 @@ void Tray::slotGenericDeviceRemoved(TDEGenericDevice *dev)
 	}
 }
 
-void Tray::slotGenericHardwareEvent(TDEHardwareEvent::TDEHardwareEvent event, TQString)
-{
-	if (event == TDEHardwareEvent::HardwareListModified) {
-		updateDeviceTrays();
-		connectTrayDeviceManager();
-	}
-}
-
 void Tray::slotVPNEventHandler(TDENetworkVPNEventType::TDENetworkVPNEventType event, TQString message)
 {
 	if (event == TDENetworkVPNEventType::LoginBanner) {
@@ -1094,7 +1086,6 @@ Tray::Tray() : KSystemTray()
 	// get notified about new/removed devices
 	connect(hwdevices, TQT_SIGNAL(hardwareAdded(TDEGenericDevice*)), this, TQT_SLOT(slotGenericDeviceAdded(TDEGenericDevice*)));
 	connect(hwdevices, TQT_SIGNAL(hardwareRemoved(TDEGenericDevice*)), this, TQT_SLOT(slotGenericDeviceRemoved(TDEGenericDevice*)));
-	connect(hwdevices, TQT_SIGNAL(hardwareEvent(TDEHardwareEvent::TDEHardwareEvent, TQString)), this, TQT_SLOT(slotGenericHardwareEvent(TDEHardwareEvent::TDEHardwareEvent, TQString)));
 
 	// Notifications
 	connect(TDEGlobal::networkManager(), TQT_SIGNAL(networkConnectionStateChanged(TDENetworkGlobalManagerFlags::TDENetworkGlobalManagerFlags, TDENetworkGlobalManagerFlags::TDENetworkGlobalManagerFlags)), this, TQT_SLOT(slotStateChangedNotify(TDENetworkGlobalManagerFlags::TDENetworkGlobalManagerFlags, TDENetworkGlobalManagerFlags::TDENetworkGlobalManagerFlags)));
diff --git a/tdenetworkmanager/src/tdenetman-tray.h b/tdenetworkmanager/src/tdenetman-tray.h
index d086902..6f4fdea 100644
--- a/tdenetworkmanager/src/tdenetman-tray.h
+++ b/tdenetworkmanager/src/tdenetman-tray.h
@@ -115,7 +115,6 @@ class Tray : public KSystemTray
 	private slots:
 		void slotGenericDeviceAdded(TDEGenericDevice*);
 		void slotGenericDeviceRemoved(TDEGenericDevice*);
-		void slotGenericHardwareEvent(TDEHardwareEvent::TDEHardwareEvent, TQString);
 		void slotVPNEventHandler(TDENetworkVPNEventType::TDENetworkVPNEventType event, TQString message);
 
 	private:
diff --git a/tdenetworkmanager/src/tdenetman-wireless_device_tray.cpp b/tdenetworkmanager/src/tdenetman-wireless_device_tray.cpp
index 2d4eccb..9b855ec 100644
--- a/tdenetworkmanager/src/tdenetman-wireless_device_tray.cpp
+++ b/tdenetworkmanager/src/tdenetman-wireless_device_tray.cpp
@@ -487,9 +487,6 @@ WirelessDeviceTray::WirelessDeviceTray (TQString dev, KSystemTray * parent, cons
 	setPixmapForState(TDENetworkConnectionStatus::Disconnected, "wireless");
 	setPixmapForState(TDENetworkConnectionStatus::Connected, "nm_signal_50");
 
-	// Listen for hardware change events
-	connect(hwdevices, TQT_SIGNAL(hardwareEvent(TDEHardwareEvent::TDEHardwareEvent, TQString)), this, TQT_SLOT(tdeHardwareEventHandler(TDEHardwareEvent::TDEHardwareEvent, TQString)));
-
 	// Trigger an initial hardware information update
 	tdeHardwareEventHandler(TDEHardwareEvent::HardwareListModified, TQString::null);
 }
-- 
2.30.2

